MCprep_addon/spawner/mcmodel.py

```diff
@@ -69,52 +69,71 @@ def rotate_around(
-		rotate_around(
-			rot_angle, [elm_to[0], elm_from[1], elm_from[2]], rot_origin, rot_axis),
-		rotate_around(
-			rot_angle, [elm_from[0], elm_from[1], elm_from[2]], rot_origin, rot_axis),
-		rotate_around(
-			rot_angle, [elm_from[0], elm_to[1], elm_to[2]], rot_origin, rot_axis),
-		rotate_around(
-			rot_angle, [elm_to[0], elm_to[1], elm_to[2]], rot_origin, rot_axis),
-		rotate_around(
-			rot_angle, [elm_to[0], elm_from[1], elm_to[2]], rot_origin, rot_axis),
-		rotate_around(
-			rot_angle, [elm_from[0], elm_from[1], elm_to[2]], rot_origin, rot_axis),
-	]
-
-	edges = []
-	faces = [
-		[0, 1, 2, 3],
-		[5, 4, 7, 6],
-		[1, 0, 4, 5],
-		[7, 6, 2, 3],
-		[4, 0, 3, 7],
-		[1, 5, 6, 2]]
-
-	return [verts, edges, faces]
-
-
-def add_material(name: str="material", path: str="") -> Material:
+		elm_from: VectorType=[0, 0, 0],
+		elm_to: VectorType=[16, 16, 16],
+		rot_origin: VectorType=[8, 8, 8],
+		rot_axis: str='y',
+		rot_angle: float=0) -> list:
+		"""Calculates and defines the verts, edge, and faces that to create."""
+		verts = [
+			rotate_around(
+				rot_angle, [elm_from[0], elm_to[1], elm_from[2]], rot_origin, rot_axis),
+			rotate_around(
+				rot_angle, [elm_to[0], elm_to[1], elm_from[2]], rot_origin, rot_axis),
+			rotate_around(
+				rot_angle, [elm_to[0], elm_from[1], elm_from[2]], rot_origin, rot_axis),
+			rotate_around(
+				rot_angle, [elm_from[0], elm_from[1], elm_from[2]], rot_origin, rot_axis),
+			rotate_around(
+				rot_angle, [elm_from[0], elm_to[1], elm_to[2]], rot_origin, rot_axis),
+			rotate_around(
+				rot_angle, [elm_to[0], elm_to[1], elm_to[2]], rot_origin, rot_axis),
+			rotate_around(
+				rot_angle, [elm_to[0], elm_from[1], elm_to[2]], rot_origin, rot_axis),
+			rotate_around(
+				rot_angle, [elm_from[0], elm_from[1], elm_to[2]], rot_origin, rot_axis),
+		]
+	
+		edges = []
+		faces = [
+			[0, 1, 2, 3], # north
+			[5, 4, 7, 6], # south
+			[1, 0, 4, 5], # up
+			[7, 6, 2, 3], # down
+			[4, 0, 3, 7], # west
+			[1, 5, 6, 2]] # east
+
+		return verts, edges, faces
+
+def add_material(name: str="material", path: str="", use_name: bool= False) -> Material:
 	"""Creates a simple material with an image texture from path."""
+	engine = bpy.context.scene.render.engine
 	cur_mats = list(bpy.data.materials)
-	res = bpy.ops.mcprep.load_material(filepath=path, skipUsage=True)
-	if res != {'FINISHED'}:
-		env.log("Failed to generate material as specified")
+	mat, err = generate.generate_base_material(bpy.context, name, path, False)
+	if not (mat is None and err):
+		passes = generate.get_textures(mat)
+		for pass_name in passes:
+			if pass_name != "diffuse":
+				passes[pass_name] = None
+		if engine == 'CYCLES' or engine == 'BLENDER_EEVEE':
+			options = generate.PrepOptions(
+				passes, 
+				False, 
+				True, 
+				False, 
+				generate.PackFormat.SIMPLE, 
+				False, 
+				False # This is for an option set in matprep_cycles
+			)
+			res = generate.matprep_cycles(
+				mat=mat,
+				options=options
+			)
```

Comment:
That's… a lot of uncommented code with special behavior. I'm honestly confused at what I'm looking at here. Please comment this.